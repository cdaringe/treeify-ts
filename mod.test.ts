import { assertSnapshot } from "@std/testing/snapshot";
import { asTree } from "./mod.ts";
import { assert } from "@std/assert";

Deno.test({
    name: 'basic tree',
    fn(t) {
        const obj = {
            a: 1
        };
        const tree = asTree(obj);
        assertSnapshot(t, tree)
    }
})

Deno.test({
  name: 'nested tree',
  fn(t) {
      const obj = {
          a: 1,
          b: {
              c: 2
          }
      };
      const tree = asTree(obj);
      assertSnapshot(t, tree)
  }
})

Deno.test({
  name: 'self-referential tree',
  fn(t) {
    const b = { c: 2, z: 'z' };
      const obj = {
          a: 1,
          b: b,
          d: b
      };
      const tree = asTree(obj);
      assertSnapshot(t, tree)
  }
})

Deno.test({
  name: 'multi-type tree',
  fn(t) {
      const obj = {
          a: 1,
          b: 'b',
          c: {
            d: true,
            e: null,
          }
      };
      const tree = asTree(obj);
      assertSnapshot(t, tree)
  }
})

Deno.test({
  name: 'show values tree',
  fn(t) {
      const obj = {
          a: 1,
          b: 'b',
          c: {
            d: true,
            e: null,
          }
      };
      const tree = asTree(obj, true);
      assertSnapshot(t, tree)
  }
})


Deno.test({
  name: 'snapshot-bug-add-frivolous-test-to-force-all-serialized',
  fn(_t) {
    assert(true)
  }
})
