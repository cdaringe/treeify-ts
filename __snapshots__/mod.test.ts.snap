export const snapshot = {};

snapshot[`basic tree 1`] = `
"└─ a
"
`;

snapshot[`nested tree 1`] = `
"├─ a
└─ b
   └─ c
"
`;

snapshot[`self-referential tree 1`] = `
"├─ a
├─ b
│  ├─ c
│  └─ z
└─ d
   ├─ c
   └─ z
"
`;

snapshot[`multi-type tree 1`] = `
"├─ a
├─ b
└─ c
   ├─ d
   └─ e
"
`;

snapshot[`show values tree 1`] = `
"├─ a: 1
├─ b: b
└─ c
   ├─ d: true
   └─ e: null
"
`;
